osf/metadata/osf_gathering.py

```diff
@@ -0,0 +1,518 @@
+        basket = pls_gather_item_metadata(osfguid.referent)
+        serializer = serializer_class(serializer_config)
+        return SerializedMetadataFile(
+            serializer.mediatype,
+            serializer.filename(osfguid._id),
+            serializer.serialize(basket),
+        )
+
+def pls_gather_item_metadata(osf_item) -> gather.Basket:
+    '''for when you just want a basket of rdf metadata about a thing
+
+    @osf_item: the thing (osf model instance or 5-ish character guid string)
+    '''
+    focus = get_osf_focus(osf_item)
+    basket = gather.Basket(focus)
+    if focus.rdftype == OSF.File:
+        basket.pls_gather_by_map(OSF_FILE_METADATA)
+    elif focus.rdftype in (OSF.Project, OSF.Component, OSF.Preprint, OSF.Registration):
+        basket.pls_gather_by_map(OSF_NODELIKE_METADATA)
+    else:
+        basket.pls_gather_by_map(OSF_COMMON_METADATA)
+    return basket
+
+
+##### END "public" api #####
+
+
+##### BEGIN osfmap #####
+
+OSF_AGENT_REFERENCE = {
+    DCT.identifier: None,
+    FOAF.name: None,
+    OSF.affiliated_institution: None,
+}
+
+OSF_COMMON_METADATA = {
+    DCT.identifier: None,
+    OWL.sameAs: None,
+    DCT.type: None,
+    DCT.title: None,
+    DCT.description: None,
+    DCT.created: None,
+    DCT.available: None,
+    DCT.modified: None,
+    DCT.dateSubmitted: None,
+    DCT.dateAccepted: None,
+    DCT.dateCopyrighted: None,
+    DCT.creator: OSF_AGENT_REFERENCE,
+    DCT.contributor: OSF_AGENT_REFERENCE,
+    DCT.language: None,
+    DCT.relation: None,
+    DCT.rightsHolder: None,
+    DCT.rights: None,
+    DCT.subject: None,
+    OSF.keyword: None,
+    OSF.affiliated_institution: None,
+    OSF.funder: None,
+}
+
+OSF_FILE_METADATA = {
+    **OSF_COMMON_METADATA,
+    DCT.hasVersion: {
+        DCT.creator: OSF_AGENT_REFERENCE,
+    },
+    OSF.has_content: None,
+    OSF.file_name: None,
+    OSF.file_path: None,
+    DCT.isPartOf: OSF_COMMON_METADATA,
+}
+
+OSF_NODELIKE_METADATA = {
+    **OSF_COMMON_METADATA,
+    DCT.isPartOf: OSF_COMMON_METADATA,
+    DCT.hasPart: OSF_COMMON_METADATA,
+    OSF.has_file: OSF_FILE_METADATA,
+}
+
+OSF_ARTIFACT_PREDICATES = {
+    ArtifactTypes.DATA: OSF.data_resource,
+    ArtifactTypes.ANALYTIC_CODE: OSF.analytic_code_resource,
+    ArtifactTypes.MATERIALS: OSF.materials_resource,
+    ArtifactTypes.PAPERS: OSF.papers_resource,
+    ArtifactTypes.SUPPLEMENTS: OSF.supplements_resource,
+}
+
+##### END osfmap #####
+
+
+##### BEGIN osf-specific utils #####
+
+def get_osf_focus(osf_item):
+    if isinstance(osf_item, str):
+        osf_item = osfdb.base.coerce_guid(osf_item).referent
+    iri = osf_iri(osf_item)
+    rdftype = get_rdf_type(osf_item)
+    focus = gather.Focus(
+        iri=iri,
+        rdftype=rdftype,
+        dbmodel=osf_item,
+    )
+    try:
+        focus.guid_metadata_record = osfdb.GuidMetadataRecord.objects.for_guid(osf_item)
+    except Exception:
+        pass
```

Comment:
What's happening here? Is the metadata record unnecessary and thus the focus can be returned if there's an exception? Is it possible to tighten this up a bit to an expected exception, or are there just too many potential exceptions that are not that important to track separately?

Reply:
yeah there's no guarantee the given item has a guid or a guid-metadata record (e.g. `FileVersion` can be a focus), so if anything goes wrong here the `focus` is still fine to return -- but yes, can and should tighten up the exception expectation, good point

(...tho also in writing this function i came to regret attaching models directly to the focus, want to make `Focus` a clean `(iri, rdftype)` namedtuple and use a separate mechanism for gatherers to get/share cached models by focus...)